source/Locales/BooleanAlgebra.lagda

```diff
@@ -0,0 +1,719 @@
+
+\end{code}
+
+\begin{code}
+
+is-lattice-homomorphism : (B : BooleanAlgebra 𝓤′ 𝓥′) (L : Frame 𝓤 𝓥 𝓦)
+                        → (⟪ B ⟫ → ⟨ L ⟩) → Ω (𝓤′ ⊔ 𝓤)
+is-lattice-homomorphism {𝓤′} {𝓥′} {𝓤} {𝓥} B L η = β ∧ γ ∧ δ ∧ ϵ
+ where
+  iss : is-set ⟨ L ⟩
+  iss = carrier-of-[ poset-of L ]-is-set
+
+  β : Ω 𝓤
+  β = η ⊤[ B ] ＝[ iss ]＝ 𝟏[ L ]
+
+  γ : Ω (𝓤′ ⊔ 𝓤)
+  γ = Ɐ x ∶ ⟪ B ⟫ , Ɐ y ∶ ⟪ B ⟫ , η (x ⋏[ B ] y) ＝[ iss ]＝ η x ∧[ L ] η y
+
+  δ : Ω 𝓤
+  δ = η ⊥[ B ] ＝[ iss ]＝ 𝟎[ L ]
+
+  ϵ : Ω (𝓤′ ⊔ 𝓤)
+  ϵ = Ɐ x ∶ ⟪ B ⟫ , Ɐ y ∶ ⟪ B ⟫ , η (x ⋎[ B ] y) ＝[ iss ]＝ η x ∨[ L ] η y
+
+lattice-homomorphisms-are-monotone : (B : BooleanAlgebra 𝓤′ 𝓥′) (L : Frame 𝓤 𝓥 𝓦)
+                                    → (h : ⟪ B ⟫ → ⟨ L ⟩)
+                                    → is-lattice-homomorphism B L h holds
+                                    → (x y : ⟪ B ⟫)
+                                    → (x ≤[ poset-of-ba B ] y) holds
+                                    → (h x ≤[ poset-of L ] h y) holds
+lattice-homomorphisms-are-monotone B L h (β , γ , _) x y p =
+ h x ＝⟨ † ⁻¹ ⟩ₚ h x ∧[ L ] h y ≤⟨ ∧[ L ]-lower₂ (h x) (h y) ⟩ h y ■
+  where
+   open PosetReasoning (poset-of L)
+
+   ‡ : x ⋏[ B ] y ＝ x
+   ‡ = ≤-is-antisymmetric (poset-of-ba B)
+        (⋏[ B ]-is-lower₁ x y)
+        (⋏[ B ]-is-greatest (≤-is-reflexive (poset-of-ba B) x) p)
+
+   † : h x ∧[ L ] h y ＝ h x
+   † = h x ∧[ L ] h y      ＝⟨ γ x y ⁻¹  ⟩
+       h (x ⋏[ B ] y)      ＝⟨ ap h ‡    ⟩
+       h x                 ∎
+
+is-embedding : (B : BooleanAlgebra 𝓤′ 𝓥′) (L : Frame 𝓤 𝓥 𝓦)
+             → (⟪ B ⟫ → ⟨ L ⟩) → Ω (𝓤′ ⊔ 𝓤)
+is-embedding {𝓤′} {𝓥′} {𝓤} {𝓥} {𝓦} B L η =
+ ι ∧ is-lattice-homomorphism B L η
+  where
+   iss : is-set ⟨ L ⟩
+   iss = carrier-of-[ poset-of L ]-is-set
+
+   iss₀ : is-set ⟪ B ⟫
+   iss₀ = carrier-of-[ poset-of-ba B ]-is-set
+
+   ι : Ω (𝓤′ ⊔ 𝓤)
+   ι = Ɐ x ∶ ⟪ B ⟫ , Ɐ y ∶ ⟪ B ⟫ , (η x ＝[ iss ]＝ η y) ⇒ (x ＝[ iss₀ ]＝ y)
+
+embedding-preserves-meets : (B : BooleanAlgebra 𝓤′ 𝓥′) (L : Frame 𝓤 𝓥 𝓦)
+                          → (η : ⟪ B ⟫ → ⟨ L ⟩)
+                          → is-embedding B L η holds
+                          → (x y : ⟪ B ⟫) → η (x ⋏[ B ] y) ＝ η x ∧[ L ] η y
+embedding-preserves-meets B L η (_ , (_ , ξ , _)) = ξ
+
+embedding-injective : (B : BooleanAlgebra 𝓤′ 𝓥′) (L : Frame 𝓤 𝓥 𝓦)
+                    → (η : ⟪ B ⟫ → ⟨ L ⟩)
+                    → is-embedding B L η holds
+                    → (x y : ⟪ B ⟫) → η x ＝ η y → x ＝ y
+embedding-injective B L η (ι , _) = ι
+
+is-spectral′ : (B : BooleanAlgebra 𝓤′ 𝓥′) (L : Frame 𝓤 𝓥 𝓦)
+            → (f : ⟪ B ⟫ → ⟨ L ⟩) → Ω (𝓤′ ⊔ 𝓤 ⊔ 𝓥 ⊔ 𝓦 ⁺)
+is-spectral′ B L f = Ɐ x ∶ ⟪ B ⟫ , is-compact-open L (f x)
+
+\end{code}
+
+\begin{code}
+
+_is-sublattice-of_ : BooleanAlgebra 𝓤′ 𝓥′ → Frame 𝓤 𝓥 𝓦 → Ω (𝓤′ ⊔ 𝓤)
+_is-sublattice-of_ B L = Ǝ η ∶ (⟪ B ⟫ → ⟨ L ⟩) , is-embedding B L η holds
+
+\end{code}
+
+\begin{code}
+
+embedding-is-order-isomorphism : (B : BooleanAlgebra 𝓤′ 𝓥′) (L : Frame 𝓤 𝓥 𝓦)
```

Comment:
This is not what an order isomorphism is. What you have here is "order preserving and reflecting".

Reply:
So rename the function, please.

Reply:
Renamed to `embedding-preserves-and-reflects-order`.